docs/advanced-features/react-18/server-components.md

```diff
@@ -0,0 +1,126 @@
+
+For client components, append `.client.js` to the filename. For example, `./components/avatar.client.js`.
+
+You can then import other server or client components from any server component. Note: a server component **can not** be imported by a client component. Components without "server/client" extensions will be treated as "universal components" and can be used and rendered by both sides, depending on where it is imported. For example:
+
+```jsx
+// pages/home.server.js
+
+import { Suspense } from 'react'
+
+import Profile from '../components/profile.server'
+import Content from '../components/content.client'
+
+export default function Home() {
+  return (
+    <div>
+      <h1>Welcome to React Server Components</h1>
+      <Suspense fallback={'Loading...'}>
+        <Profile />
+      </Suspense>
+      <Content />
+    </div>
+  )
+}
+```
+
+The `<Home>` and `<Profile>` components will always be server-side rendered and streamed to the client, and will not be included by the client runtime. However, `<Content>` will still be hydrated on the client-side, like normal React components.
+
+> Notice that make sure you're using default imports for server and client components at the moment. The the support of named exports are working in progress!
+
+To see a full example, check out the [vercel/next-rsc-demo demo](https://github.com/vercel/next-rsc-demo).
+
+## Supported Next.js APIs
+
+### `next/link` and `next/image`
+
+You can use `next/link` and `next/image` like before and they will be treated as client components to keep the interaction on client side.
+
+### `next/document`
+
+If you have custom `_document`, you have to change your `_document` to a functional component like below to use server components. Or if you don't have any, Next.js will provide a functional fallback `_document` component.
+
+```jsx
+// pages/_document.js
+import { Html, Head, Main, NextScript } from 'next/document'
+
+export default function Document() {
+  return (
+    <Html>
+      <Head />
+      <body>
+        <Main />
+        <NextScript />
+      </body>
+    </Html>
+  )
+}
+```
+
+### `next/app`
+
+If you're using `_app.js`, the usage is the same as [Custom App](/docs/advanced-features/custom-app).
+If you're using `_app.server.js` as a server component, the signature is changed as below where it only receives the `children` prop as elements. You can wrap any other client or server components around `children` to customize the layout of your app.
+
+```js
+// pages/_app.server.js
+export default function App({ children }) {
+  return children
+}
+```
+
+### Routing
+
+Basic routes with path and queries, dynamic routes are supported as well. If you need to access the router in server components(`.server.js`), They will receive `router` instance as a prop so that you could directly accessing them without using `useRouter()` hook.
```

Comment:
```suggestion
Both basic routes with path and queries and dynamic routes are supported. If you need to access the router in server components(`.server.js`), they will receive `router` instance as a prop so that you can directly access them without using the `useRouter()` hook.
```